schema/chat_room.proto

```diff
@@ -0,0 +1,343 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "google/longrunning/operations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/error_details.proto";
+
+package google.showcase.v1alpha3;
+
+option go_package = "github.com/googleapis/gapic-showcase/server/genproto";
+option java_package = "com.google.showcase.v1alpha3";
+option java_multiple_files = true;
+
+option (google.api.metadata) = {
+  product_uri: "https://github.com/googleapis/gapic-showcase"
+};
+
+service ChatRoom {
+  rpc CreateRoom(CreateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      post: "/v1alpha3/rooms"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["display_name", "description"]
+    };
+  }
+
+  rpc GetRoom(GetRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  rpc UpdateRoom(UpdateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{name=rooms/*}"
+      body: "*"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name", "room.display_name"]
+      additional_signatures: {
+        fields: ["name", "room.display_name", "room.description"]
+      }
+    };
+  }
+
+  rpc DeleteRoom(DeleteRoomRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  rpc ListRooms(ListRoomsRequest) returns (ListRoomsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/rooms"
+    };
+  }
+
+  rpc StreamRoom(StreamRoomRequest) returns (stream Message) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{name=rooms/*}:streamRoom"
+      body: "*"
+      additional_bindings: {
+        post: "/v1alpha3/{name=global}:streamRoom"
+        body: "*"
+      }
+    };
+  }
+
+  rpc SearchAll(SearchAllRequest) returns (google.longrunning.Operation) {
```

Comment:
Search all...what? Needs a noun. (It probably also does not need "all".)

Assuming this does what I think you intend it to do, it should be `rpc SearchMessages`.

Reply:
Changed to SearchMessages